Sources/BraintreeCore/BTHTTP.m

```diff
@@ -156,15 +156,19 @@ - (void)httpRequestWithCaching:(NSString *)method path:(NSString *)aPath paramet
             [[NSURLCache sharedURLCache] removeAllCachedResponses];
             cachedResponse = nil;
         }
-
-        if (cachedResponse != nil) {
-            [self handleRequestCompletion:cachedResponse.data request:nil shouldCache:NO response:cachedResponse.response error:nil completionBlock:completionBlock];
-        } else {
-            NSURLSessionTask *task = [self.session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-                [self handleRequestCompletion:data request:request shouldCache:YES response:response error:error completionBlock:completionBlock];
-            }];
-            [task resume];
-        }
+        
+        // The increase in speed of API calls with cached configuration caused an increase in "network connection lost" errors.
+        // Adding this delay allows us to throttle the network requests slightly to reduce load on the servers and decrease connection lost errors.
+        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 0.1 * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
+            if (cachedResponse != nil) {
+                [self handleRequestCompletion:cachedResponse.data request:nil shouldCache:NO response:cachedResponse.response error:nil completionBlock:completionBlock];
+            } else {
+                NSURLSessionTask *task = [self.session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                    [self handleRequestCompletion:data request:request shouldCache:YES response:response error:error completionBlock:completionBlock];
+                }];
+                [task resume];
+            }
+        });
```

Comment:
Is the `httpRequestWithCaching:` only used for `BTConfiguration` requests?

Reply:
Yep! We only want to cache configuration requests (not anything else)

Reply:
Is the overall solution to prevent multiple BT API requests within a 1 second timeframe?

Reply:
Yeah, essentially we are throttling this request by 0.1 seconds

Reply:
Oh aight Analytics HTTP requests happen as soon as they're triggered on iOS that is rough.

Reply:
Don't mean to nit I genuinely have questions lol. Did we choose 0.1 because it worked best in testing?

Reply:
Yeah we tried 0.05 and 0.075 but we still saw an increased number of Network connection lost errors. 0.1 seems to be the least amount of time that still improved the errors.